libyul/AsmAnalysis.cpp

```diff
@@ -442,15 +442,20 @@ bool AsmAnalyzer::operator()(Switch const& _switch)
 		if (_case.value)
 		{
 			int const initialStackHeight = m_stackHeight;
+			bool isCaseValueValid = true;
 			// We cannot use "expectExpression" here because *_case.value is not a
 			// Statement and would be converted to a Statement otherwise.
 			if (!(*this)(*_case.value))
+			{
+				isCaseValueValid = false;
 				success = false;
+			}
 			expectDeposit(1, initialStackHeight, _case.value->location);
 			m_stackHeight--;
 
+			// If the case value is not valid, we should not insert it into cases.
```

Comment:
Please add an assertion like `yulAssert(isCaseValueValid || m_errorReporter.hasErrors()`, just in case.

Also please add a test case in `test/libyul/Parser.cpp`.

Reply:
Fixed. And also add a test case for this PR.